src/mlpack/core/optimizers/sdp/lrsdp_function.hpp

```diff
@@ -57,110 +73,86 @@ class LRSDPFunction
                           arma::mat& gradient) const;
 
   //! Get the number of sparse constraints in the LRSDP.
-  size_t NumSparseConstraints() const { return sparseB.n_elem; }
+  size_t NumSparseConstraints() const { return sdp.NumSparseConstraints(); }
```

Comment:
I think that is better, but I didn't want to refactor existing code :smile: 

I'll make the changes